src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2053,6 +2148,73 @@ mesh::generate_index_for_single_domain(const Node &mesh,
 
 
 
+//-------------------------------------------------------------------------
+// blueprint tests and converters for one-dimensional meshes
+//-------------------------------------------------------------------------
+
+//-----------------------------------------------------------------------------
+bool
+mesh::can_generate_strip(const Node &mesh,
+                         const std::string & topo_name,
+                         Node &info)
+{
+    const std::string protocol = "can_generate_strip";
+    bool res = true;
+    info.reset();
+
+    // The topology must be 1D and cannot be points
+    const Node& topo = mesh["topologies"][topo_name];
+    const Node& coordset = mesh["coordsets"][topo["coordset"].as_string()];
+    index_t cs_dim = mesh::utils::coordset::dims(coordset);
+    std::string topo_type = topo["type"].as_string();
+    if (!(cs_dim == 1 && topo_type != "points"))
+    {
+        log::error(info, protocol, "topology type is points or coordset dimension other than 1");
+        res = false;
+    }
+
+    // Only element-associated fields allowed (no vertex-fields)
+    // TODO Relax this requirement?
+    NodeConstIterator fitr = mesh["fields"].children();
+    while(fitr.has_next())
+    {
+        const Node &f = fitr.next();
+        // This method requires fields to have "association" == "element".
+        if (!f.has_child("association") || f["association"].as_string() != "element")
+        {
+            log::error(info,
+                       protocol,
+                       "fields[" + log::quote(fitr.name()) + "/association] != element");
+            res = false;
+        }
+    }
+
+    return res;
+}
+
+//-------------------------------------------------------------------------
+void
+mesh::generate_strip(conduit::Node &mesh,
+                     std::string src_topo_name,
+                     std::string dst_topo_name)
+{
+
+    const Node& src_topo = mesh["topologies"][src_topo_name];
+    const Node& src_coordset = mesh["coordsets"][src_topo["coordset"].as_string()];
+    Node dst_topo, dst_coordset;
+
+    mesh::topology::generate_strip(src_topo, dst_topo);
+    mesh::coordset::generate_strip(src_coordset, dst_coordset);
+
+    mesh["topologies"][dst_topo_name] = dst_topo;
+    mesh["coordsets"][dst_topo_name] = dst_coordset;
+
+
+    // Flag this as a 1D strip mesh
+    mesh["private_nonblueprint/carterflags/oneD_strip"] = 1;
+}
```

Comment:
Should we put this flag in the conduit carter interface instead of here?

Reply:
Well, it should go somewhere.  I'll remove it.